packages/terra-form/src/NumberField.jsx

```diff
@@ -0,0 +1,125 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import Field from './Field';
+import Input from './Input';
+
+const propTypes = {
+  /**
+   * The defaultValue of the input element
+   */
+  defaultValue: PropTypes.number,
+  /**
+   * Error message for when the input is invalid
+   */
+  error: PropTypes.node,
+  /**
+   *  Help element to display with the input
+   */
+  help: PropTypes.node,
+  /**
+   * Additional attributes to attach to the number input
+   */
+  inputAttrs: PropTypes.object,
+  /**
+   * Whether the field is inline
+   */
+  isInline: PropTypes.bool,
+  /**
+   * Label of the input
+   */
+  label: PropTypes.node,
+  /**
+   * Maximum value allowed for the input
+   */
+  max: PropTypes.number,
+  /**
+   * Minimum value allowed for the input
+   */
+  min: PropTypes.number,
+  /**
+   * Name of the input attribute
+   */
+  name: PropTypes.string,
+  /**
+   * Function to trigger when user changes the input value
+   */
+  onChange: PropTypes.func,
+  /**
+   * Whether the input is required
+   */
+  required: PropTypes.bool,
+  /**
+   * Incremental steps for increasing and descreasing the value
+   */
+  step: PropTypes.number,
+  /**
+   * The value of the input element
+   */
+  value: PropTypes.oneOfType([
+    PropTypes.string,
+    PropTypes.number,
+  ]),
+};
+
+const defaultProps = {
+  defaultValue: undefined,
+  error: null,
+  help: null,
+  inputAttrs: {},
+  isInline: false,
+  label: null,
+  max: null,
+  min: null,
+  name: null,
+  onChange: undefined,
+  required: true,
+  step: null,
+  value: undefined,
+};
+
+const NumberField = (
+  {
+    error,
+    help,
+    inputAttrs,
+    isInline,
+    label,
+    max,
+    min,
+    name,
+    onChange,
+    required,
+    step,
+    value,
+    defaultValue,
+    ...customProps
+  },
+) => (
```

Comment:
I don't think we need to. All the fields should function and display the same. We do have the ability to add custom classes to these fields, which should cover the case if a consumer needs to style a field in a particular way.